src/SWP/Component/MultiTenancy/Repository/TenantRepositoryInterface.php

```diff
@@ -26,12 +26,24 @@
      * Finds the tenant by subdomain.
      *
      * @param string $subdomain The subdomain
+     * @param string $domain    The domain
      *
      * @throws \Doctrine\ORM\NonUniqueResultException
      *
      * @return TenantInterface|null The instance of TenantInterface or null
      */
-    public function findOneBySubdomain($subdomain);
+    public function findOneBySubdomainAndDomain($subdomain, $domain);
+
+    /**
+     * Finds the tenant by domain.
+     *
+     * @param string $domain The domain
+     *
+     * @throws \Doctrine\ORM\NonUniqueResultException
```

Comment:
by this u are making a dependency on Doctrine ORM which is not included into that component and it shouldnt. This interface is generic, it shouldnt rely on doctrine only, it shouldnt tell you it throws doctrine orm exception